packages/grafana-data/src/transformations/transformers/insertValues.ts

```diff
@@ -0,0 +1,141 @@
+import { DataFrame, FieldType, Field, ArrayVector } from '../..';
+
+import { map } from 'rxjs/operators';
+
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+
+const enum InsertValuesMode {
+  Interval = 'interval',
+  Threshold = 'threshold',
+}
+
+const IS_TIME_FIELD = (field: Field) => field.type === FieldType.time;
+const THRESHOLD_HOUR = 3600 * 1e3;
+
+export interface InsertValuesTransformerOptions {
+  mode: InsertValuesMode;
+  // value to insert
+  value: any;
+  // finds field to use for interval or threshold checking
+  isRefField?: (field: Field) => boolean;
```

Comment:
Transformer options can not include functions -- they can be written to JSON

Reply:
should this be field name string and if empty use type/time? and ui will have field selector?

Reply:
switched to using a string `field` option